app/src/main/java/org/xbmc/kore/utils/Task.java

```diff
@@ -0,0 +1,81 @@
+package org.xbmc.kore.utils;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Represents a computation that will eventually complete.
+ *
+ * @param <T> The type of the value being computed or awaited.
+ */
+public interface Task<T> {
+
+    /**
+     * Starts the computation, invokes a procedure when done.
+     *
+     * @param then The consumer of the value being computed.
```

Comment:
In the Sequence the value being computed is not passed to the consumer, right? If so, this seems to contradict its main usage.